include/ofxColour.h

```diff
@@ -0,0 +1,259 @@
+
+The colourspace strings used in the native styles are from ofx-native-1.5.ocio,
+which is based on the OCIO ACES Studio built-in config,
+studio-config-v2.1.0_aces-v1.3_ocio-v2.3, and stored for OFX purposes in
+openfx-native-v1.5_aces-v1.3_ocio-v2.3.h (referred to as the config header).
+Additionally, there is a scheme for cross-referencing between clips, and a set
+of "basic colourspaces", which are designed to be generic names for a family of
+colourspaces. When a basic colourspace is used, this means the host is free to
+choose any colourspace with the same encoding and reference space (scene vs
+display).
+
+The assumption is that OCIO > Full > Core > Basic, so the highest style
+supported by both host and plug-in should usually be chosen by the host.
+The chosen style must be set using this property on an image effect instance.
+*/
+#define kOfxImageEffectPropColourManagementStyle "OfxImageEffectPropColourManagementStyle"
+
+/* String used to indicate that no colour management is available. */
+#define kOfxImageEffectPropColourManagementNone "OfxImageEffectPropColourManagementNone"
+/* String used to indicate that basic colour management is available. */
+#define kOfxImageEffectPropColourManagementBasic "OfxImageEffectPropColourManagementBasic"
+/* String used to indicate that core colour management is available. */
+#define kOfxImageEffectPropColourManagementCore "OfxImageEffectPropColourManagementCore"
+/* String used to indicate that Full colour management is available. */
+#define kOfxImageEffectPropColourManagementFull "OfxImageEffectPropColourManagementFull"
+/* String used to indicate that OCIO colour management is available. */
+#define kOfxImageEffectPropColourManagementOCIO "OfxImageEffectPropColourManagementOCIO"
+
+/** @brief What native mode configs are supported?
+
+   - Type - string X N
+   - Property Set - host descriptor (read only), plugin descriptor (read/write)
+   - Valid Values - A list of config identifiers. The only currently supported
+                    value is ofx-native-v1.5_aces-v1.3_ocio-v2.3.
+
+While the API for colour management is expected to be stable, the set of
+colourspaces will evolve over time. This property must be set by both plug-ins
+and hosts specifying the list of native mode configs that are available on each
+side.
+*/
+#define kOfxImageEffectPropColourManagementAvailableConfigs "OfxImageEffectPropColourManagementAvailableConfigs"
+
+/** @brief The native colour management config to be used for this instance
+
+   - Type - string X 1
+   - Property Set - image effect instance (read only)
+   - Valid Values - any string provided by the plug-in in
+                    kOfxImageEffectPropColourManagementAvailableConfigs
+
+If kOfxImageEffectPropColourManagementStyle for the instance is a native style,
+the host must set this property to indicate the native colour management config
+the plug-in should be used to look up colourspace strings.
```

Comment:
Wouldn't the plug-in also need to set this property if it wants to use a native mode?  If more than one is available at a future point, I assume that the most recent one supported by both sides should be used?

If the native mode identified is just the basic style, then perhaps it could be omitted, for simplicity.

Reply:
The plug-in communicates the list of configs it supports via `kOfxImageEffectPropColourManagementAvailableConfigs`, then the host chooses which config it wants to use and sets that in `kOfxImageEffectPropColourManagementConfig`. This applies for the whole effect rather than specifically on the input or output side. You'd normally expect the host to choose the most recent one supported by both sides, but that might not always be the case. For example the user might have set up their colour pipeline based on an older version of ACES. This is why it's left for the host to decide.

Also, this can't be omitted in the basic style, because the basic colourspaces are defined in the config header and we might add new ones in future.